### PR TITLE
Update rustc-ap-syntax to 606.0.0

### LINKED ISSUE #1074: Build failure on nightly Rust
Hi,

`cargo +nightly install racer` fails with the following:

```
error[E0046]: not all trait items implemented, missing: `mixed_site`
   --> /home/jakob/.cargo/registry/src/github.com-1ecc6299db9ec823/rustc-ap-syntax-583.0.0/ext/proc_macro_server.rs:657:1
    |
657 | impl server::Span for Rustc<'_> {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ missing `mixed_site` in implementation
    |
    = note: `mixed_site` from trait: `fn(&mut Self) -> <Self as proc_macro::bridge::server::Types>::Span`
```

This is on `rustc 1.40.0-nightly (2e7244807 2019-10-04)`, and the version of `rustc-ap-syntax` being brought in is `v583.0.0`.

### PR DESCRIPTION
**Pull request by TsarFox**
Resolves #1074

### CODE REVIEW

**Comment by kngwyu** [comment]
Thank you!

File: src/racer/lib.rs [hunk 1]
```diff
@@ -11,6 +11,8 @@ extern crate bitflags;
 #[macro_use]
 extern crate derive_more;
 
+extern crate rustc_ap_syntax as syntax;
```
Comment (kngwyu) [inline src/racer/lib.rs]: Do we really need this?

Comment (TsarFox) [reply src/racer/lib.rs]: Yes. Without it, building complains about "use of undeclared type or module `syntax`"

Comment (TsarFox) [reply src/racer/lib.rs]: I suppose the alternative is changing every reference of `syntax` to `rustc_ap_syntax`, but I thought this was a little less intrusive

Comment (kngwyu) [reply src/racer/lib.rs]: Ah, OK.
Then, how about setting it in toml
```toml
[dependencies.syntax]
package = "rustc-ap-syntax"
version = "606.0.0"
```
like [rustfmt does](https://github.com/rust-lang/rustfmt/pull/3835/files)?

Comment (TsarFox) [reply src/racer/lib.rs]: Didn't know you could do that, thanks :]

I agree that's much nicer. Changed.

**Comment by kngwyu** [comment]
Thank you for your quick fix, again.

**Comment by TsarFox** [comment]
My pleasure :)

**Comment by ghedo** [comment]
@kngwyu is there an ETA for a new release with this fix? Just so https://github.com/rust-lang/rust/issues/65084 and https://github.com/rust-lang/rust/issues/65085 can be unblocked.

**Comment by kngwyu** [comment]
@ghedo 
Released 2.1.28 now.
Thank you for letting me know that.

**Pull request merged**